src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -1387,10 +1582,12 @@ bool rasterImage(SwSurface* surface, SwImage* image, const Matrix* transform, co
             return _rasterImageRle(surface, image->rle, image->data, image->w, image->h);
         } else {
             if (translucent) {
-                if (isUpScaling) return _rasterTranslucentUpScaleImageRle(surface, image->rle, image->data, image->w, image->h, opacity, &invTransform);
+                if (scaling > 1.0f) return _rasterTranslucentUpScaleImageRle(surface, image->rle, image->data, image->w, image->h, opacity, &invTransform);
```

Comment:
I'd rather compare floats using epsilon, but as you wish. In the worst case scenario an unnecessary scaling will be done

Reply:
Fixed to use FLT_EPSILON.